plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -371,6 +549,49 @@ private function build_query(): void {
 		$this->sql = "SELECT $found_rows $distinct $fields FROM $orders_table $join WHERE $where $groupby $orderby $limits";
 	}
 
+	/**
+	 * JOINs the main orders table with another table.
+	 *
+	 * @param string  $table      Table name (including prefix).
+	 * @param string  $alias      Table alias to use. Defaults to $table.
+	 * @param string  $on         ON clause. Defaults to "wc_orders.id = {$alias}.order_id".
+	 * @param boolean $alias_once If TRUE, table won't be JOIN'ed again if already JOIN'ed.
+	 * @return void
+	 * @throws \Exception When an error occurs, such as trying to re-use an alias with $alias_once = FALSE.
+	 */
+	public function join( string $table, string $alias = '', string $on = '', bool $alias_once = false ) {
```

Comment:
does this need to be `public`? i think we should mark it `private` since `$on` is an unsecured param. (I see most other functions are private, so maybe this one does have some requirement)

Reply:
I was thinking other objects possibly extending querying capabilities (support for other query vars, etc.) might want to use this method (similarly to how `where()` is public) but I understand the `$on` arg is problematic. I'm going to change that to `private` and we can open it again depending on how we decide to implement extensibility.